documents/MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,1081 @@
+
+V2 SDK supports different connection types. Given the same input parameters as in the V1 example above, the most suitable method to create an MQTT5 client will [beNewMqtt5ClientBuilderWithMtlsFromPath](https://aws.github.io/aws-iot-device-sdk-cpp-v2/class_aws_1_1_iot_1_1_mqtt5_client_builder.html#ab595bbc50e08b9d2f78f62e9efeafd65).
+
+```
+util::String clientEndpoint = "<prefix>-ats.iot.<region>.amazonaws.com";
+util::String certificateFile = "<certificate file>";  // X.509 based certificate file
+util::String privateKeyFile = "<private key file>";   // PEM encoded private key file
+uint32_t clientPort = <port number>
+
+std::shared_ptr<Aws::Iot::Mqtt5ClientBuilder>builder(
+    Aws::Iot::Mqtt5ClientBuilder::NewMqtt5ClientBuilderWithMtlsFromPath(
+        clientEndpoint,
+        certificateFie,
+        privateKeyFile));
+std::shared_ptr<Mqtt5::ConnectPacket> connectOptions =
+        std::make_shared<Mqtt5::ConnectPacket>();
+util::String clientId = "<unique client id>";
+connectOptions->WithClientId(clientId);
+builder->WithConnectOptions(connectOptions);
+builder->WithPort(clientPort);
+std::shared_ptr<Aws::Crt::Mqtt5::Mqtt5Client> client = builder->Build();
+```
+
+Refer to the [Connection Types and Features](https://quip-amazon.com/bbcuA13anxJa#temp:C:YWR0b9b69e08b72487dbe620342f) section for other connection types supported by V2 SDK.
+
+### Connection Types and Features
+
+V1 SDK supports two types of connections to connect to the AWS IoT service: MQTT with X.509 certificate and MQTT over Secure WebSocket with SigV4 authentication.
+
+V2 SDK adds a collection of connection types and cryptography formats (e.g. [PKCS #11](https://en.wikipedia.org/wiki/PKCS_11) and [Custom Authorizer](https://docs.aws.amazon.com/iot/latest/developerguide/custom-authentication.html)), credential providers (e.g. [Amazon Cognito](https://aws.amazon.com/cognito/) and [Windows Certificate Store](https://learn.microsoft.com/en-us/windows-hardware/drivers/install/certificate-stores)), and other connection-related features.
+Refer to the “[Connecting To AWS IoT Core](https://github.com/aws/aws-iot-device-sdk-cpp-v2/blob/main/documents/MQTT5_Userguide.md#connecting-to-aws-iot-core)” section of the MQTT5 user guide for detailed information and code snippets on each connection type and connection feature.
+
+|Connection Type/Feature	|V1 SDK	|V2 SDK	|User guide section	|	|
+|---	|---	|---	|---	|---	|
+|	|	|	|	|	|
+|MQTT over Secure WebSocket with AWS SigV4 authentication	|✔	|✔	|	|	|
+|MQTT (over TLS 1.2) with X.509 certificate based mutual authentication	|✘	|✔	|	|X.509	|
+|MQTT with PKCS12 Method	|✔*	|✔	|	|Container for X.509	|
+|MQTT with Custom Key Operation Method	|✔*	|✔	|	|X.509	|
+|MQTT with Custom Authorizer Method	|✔*	|✔	|	|	|
+|MQTT with Windows Certificate Store Method	|✔*	|✔	|	|X.509	|
+|MQTT with PKCS11 Method	|✘	|✔	|	|X.509 plus other formats	|
+|Websocket Connection with Cognito Authentication Method	|✘	|✔	|	|	|
+|HTTP Proxy	|✔**	|✔	|	|	|
```

Comment:
Debatable: Use colors for checks and crosses. I used latex for that: `$${\Large\color{green}&#10004}$$` and `$${\Large\color{orange}&#10004\*}$$`. See in the [java pr](https://github.com/aws/aws-iot-device-sdk-java-v2/pull/533/files#diff-bcb919f53680494fc23f06bec1126f2c8b9ac21d8b01e56b06d77f9fbb31b4acR175).